### PR TITLE
New locale: DE

### LINKED ISSUE #1161: New locale: DE
My goal is to have (and share) a good German Wiktionary-based dictionary that displays well on small e-reader screens and is a little more informative (i.e., has word form, gender, hyphenation, IPA pronunciation, meaning, abbreviations, synonyms and examples). My main target format would be _StarDict_, with possible spinoff formats for Kobo _(dicthtml?)_, PocketBook (?) and Tolino _(quickdic)_.

Too bad _pyglossary_ doesn’t support R. Döffinger’s _quickdic_ format, because _Tolino_ devices use that, and we do have a rather large Tolino user base in Germany. Not everybody wants to jailbreak their device…

I currently use DE Wiktionary dumps and a rather brute-force [Rexx script](https://www.e-reader-forum.de/t/wiktionary-fuer-pocketbook.118669/page-6#post-2438607) to generate a Tabfile, which I then convert to StarDict and dicthtml formats. (See attached screenshots for how it looks in _GoldenDict_ on Linux.)

This is of course a flakey way to do it, and I’d prefer to collaborate with a more sound foundation like yours and integrate it there, also because yours gets auto-updated.

Unfortunately, the [HOWTO Add a New Locale](https://github.com/BoboTiG/ebook-reader-dict/wiki/HOWTO-Add-a-New-Locale) section in the wiki here isn’t too detailed, and I’d probably need quite a bit of help to get started. I’m especially unsure about the first two steps and the "Remove all data from the old lang."

So my questions are:

1. Would you be interested in a German dictionary that should look approximately like the screenshots show?
2. Is it possible to do, without investing too much time? (There’s a lot of other things I have to spend my time on, but I’d be willing to invest a substantial amount of time to get it started and polished a little.)
3. Is there any assistance possible in getting me set up to get the first steps done? I reckon that’d be to set up a working environment on my Linux Mint 20.3 machine, do a fork, and start adding a language "de".
4. Since I know almost nothing about Wiktionary’s internal structures, I fear the templates most. But having had a glance at your code, I think there is some expertise here…

**Screenshots:** This is how I envision it to look like. Users on [MobileRead](https://www.mobileread.com/) and the [German E-Reader Forum](https://www.e-reader-forum.de/) have been quite enthusiastic about the first version. Screenshots show the _StarDict_ version used by _GoldenDict_ on a Linux desktop.

wiktionary - GoldenDict_001

Wiktionarys - GoldenDict_001

Auswahl_194

**Links** to what exists already:

- [Rexx script](https://kaufen-ist-toll.de/download/e-reader/wiktionary2txt.rex.zip) to convert wiktionary dump to tab file
- Resulting [StarDict DE dictionary](https://kaufen-ist-toll.de/download/e-reader/MCH%20DE-DE%20Stardict.zip)
- Resulting [dicthtml DE dictionary](https://kaufen-ist-toll.de/download/e-reader/dicthtml-de-de.zip)

### PR DESCRIPTION
**Pull request by BoboTiG**
Closes #1161.

### CODE REVIEW

File: wikidict/lang/de/__init__.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+"""Deutsh language."""
+
+# Regex to find the pronunciation
+pronunciation = r"{{Lautschrift\|([^}]+)}}"
+
+# Regex to find the gender
+gender = r",\s+{{([fmn]+)}}"
+
+# Float number separator
+float_separator = "."
```
Comment (BoboTiG) [inline wikidict/lang/de/__init__.py:10]: Could you tell me the good value to put here @Moonbase59 :pray:?

File: wikidict/lang/de/__init__.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# Thousads separator
+thousands_separator = ","
```
Comment (BoboTiG) [inline wikidict/lang/de/__init__.py:13]: Could you tell me the good value to put here @Moonbase59 :pray:?

File: wikidict/lang/de/__init__.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+Wiktionary dump: {dump_date}
+
+Available files:
+
+- [Kobo]({url_kobo}) (dicthtml-{locale}-{locale}.zip)
+- [StarDict]({url_stardict}) (dict-{locale}-{locale}.zip)
+- [DictFile]({url_dictfile}) (dict-{locale}-{locale}.df)
+
+<sub>Updated on {creation_date}</sub>
+"""  # noqa
```
Comment (BoboTiG) [inline wikidict/lang/de/__init__.py:51]: @Moonbase59 if you could open a PR to translate that text, that would be awesome.

File: wikidict/lang/de/__init__.py (hunk 1 above)
Comment (Moonbase59) [reply wikidict/lang/de/__init__.py:10]: We also now have a third official gender "d" (diverse), so gender should probably be `[fmnd]`. Don’t know if it’s used already in the Wiktionary.

File: wikidict/lang/de/__init__.py (hunk 2 above)
Comment (Moonbase59) [reply wikidict/lang/de/__init__.py:13]: Both correct!

File: wikidict/lang/de/__init__.py (hunk 3 above)
Comment (Moonbase59) [reply wikidict/lang/de/__init__.py:51]: Okay, as soon as I’v got around that fork thingy…

File: wikidict/lang/de/__init__.py (hunk 2 above)
Comment (BoboTiG) [reply wikidict/lang/de/__init__.py:13]: Great!

File: wikidict/lang/de/__init__.py (hunk 3 above)
Comment (BoboTiG) [reply wikidict/lang/de/__init__.py:51]: Indeed :)

File: wikidict/lang/de/__init__.py (hunk 1 above)
Comment (BoboTiG) [reply wikidict/lang/de/__init__.py:10]: Good to know. I will let you update the regexp as an exercise :)

File: wikidict/lang/de/__init__.py (hunk 3 above)
Comment (Moonbase59) [reply wikidict/lang/de/__init__.py:51]: https://github.com/BoboTiG/ebook-reader-dict/pull/1176

**Pull request merged**